### ISSUE TITLE
Count of indices

### ISSUE THREAD

**Issue by j0hnsmith** [justwatchcom/elasticsearch_exporter#131]
There doesn't seem to be a 'total # of indices in cluster' metric, this would be useful.

**Comment by zwopir** [comment]
Hi @j0hnsmith,

I've checked the ES API endpoints we are currently querying on each prometheus scrape. They don't return the requested metric. So I would have to query another endpoint to get the metric. It would be something like `http://localhost:9200/_cat/indices?format=json&pretty`. I think the number of indices is a metric that doesn't change frequently. So adding another http call on each scrape for a more or less constant metric doesn't seem appropriate to me

**Comment by j0hnsmith** [comment]
Fair point, does depend on usage though. Is there any scope to make it optional, eg off by default but users can enable?

**Comment by zwopir** [comment]
yes, that'd be certainly possible. It's quite a quite easy task, the json returned by the endpoint pretty clear. See https://www.elastic.co/guide/en/elasticsearch/reference/current/cat.html for additional information how to change the units of the returned metrics (bytes vs. human, ...) So if we (you?) implement it we just need to make sure the queries work on all supported versions, i.e. 1.8.x, 2.x, 5.x and 6.x. and that the json looks the same. If it doesn't, I think it's not worth the effort.

So I can review a PR that optionally adds the feature and/or help with it, but working on it myself hasn't got the highest priority

**Comment by benkeil** [comment]
If you use daily indices its not a constant size.

**Comment by gavk34** [comment]
I might have time to look into this - seems the `_all/_stats` has this information no?
Should be just a matter of capturing the root(name of index) to a metric - which users can sum?
Example:
curl -s https://my-elasticsearch-url.com/_all/_stats\?format\=json\&pretty | jq '.indices'
```
"kubernetes-2018.08.27": {
    "primaries": {
    ...
```

My Go is not expert level so I wouldn't be expecting this urgently however we have use ourselves for such a metric

**Comment by j0hnsmith** [comment]
@gavk34 see the PR and further discussion in #143.

**Comment by zwopir** [comment]
thanks for the clarification, @j0hnsmith 
closing this issue

TL;DR; 
count of indices can be obtained with `count by (instance) (elasticsearch_indices_docs_primary)`
